---
title: "Differential phospho/protein analysis (Comparison t-test vs DEP) - CRC Batch 1"
author: Sophie Herbst
output:
  html_document:
    code_folding: hide
    toc: yes
    toc_depth: 4
    toc_float: yes
editor_options: 
  chunk_output_type: console
---
Analysis date: `r Sys.Date()`

# Depends on
DIPG_FirstBatch_DataProcessing Script

```{r}
load("../Data/Cache/Xenografts_Batch1_2_DataProcessing.RData")
```

# TODO

* Do differential abudance analysis for prep batch and mass spec run

# Setup
## Load libraries and functions
```{r, echo=FALSE, warning=FALSE, message=FALSE}
set.seed(2022)
library(fgsea)
library(AnnotationDbi)
library(org.Hs.eg.db)
library(DEP)
library(SummarizedExperiment)
# library(pls)
library(mdatools)
source("../../../General/Code/Analysis_Functions.R")
source("CRC_Xenografts_Batch1_Functions.R")
```

# Analysis
## DEP
### Tyrosine all
#### Each condition vs ctrl
```{r, eval=TRUE}
Test_PhosphoData(pY_Set1_form, comparison = "E", comparison_base = "ctrl") %>% GGPlotly_Volcano_Test

data_diff_E_vs_ctrl_pY <- test_diff(pY_se_Set1, type="manual", test = "E_vs_ctrl")
dep_E_vs_ctrl_pY <- add_rejections_SH(data_diff_E_vs_ctrl_pY, alpha = 0.05, lfc = log2(1.2))
GGPlotly_Volcano(dep_E_vs_ctrl_pY, contrast = "E_vs_ctrl", 
                 add_names = TRUE,
                additional_title = "pY") 
```

```{r, fig.width=10, fig.height=10}
Return_DEP_Hits_Plots(data = pY_Set1_form, dep_E_vs_ctrl_pY, comparison = "E_vs_ctrl_diff")
```

```{r}
Test_PhosphoData(pY_Set1_form, comparison = "EC", comparison_base = "ctrl") %>% GGPlotly_Volcano_Test

data_diff_EC_vs_ctrl_pY <- test_diff(pY_se_Set1, type="manual", test = "EC_vs_ctrl")
dep_EC_vs_ctrl_pY <- add_rejections_SH(data_diff_EC_vs_ctrl_pY, alpha = 0.05, lfc = log2(1.2))
GGPlotly_Volcano(dep_EC_vs_ctrl_pY, contrast = "EC_vs_ctrl", 
                 add_names = TRUE,
                additional_title = "pY") 
```

```{r, fig.width=10, fig.height=10}
Return_DEP_Hits_Plots(data = pY_Set1_form, dep_EC_vs_ctrl_pY, comparison = "EC_vs_ctrl_diff")
```

```{r, eval=FALSE}
Plot_Enrichment_Single_Pathway(dep_EC_vs_ctrl_pY, comparison = "EC_vs_ctrl_diff", 
                               pw = "Epigenetic regulation of gene expression")
```

```{r}
Test_PhosphoData(pY_Set1_form, comparison = "EBC", comparison_base = "ctrl") %>% GGPlotly_Volcano_Test

data_diff_EBC_vs_ctrl_pY <- test_diff(pY_se_Set1, type="manual", test = "EBC_vs_ctrl")
dep_EBC_vs_ctrl_pY <- add_rejections_SH(data_diff_EBC_vs_ctrl_pY, alpha = 0.05, lfc = log2(1.2))
GGPlotly_Volcano(dep_EBC_vs_ctrl_pY, contrast = "EBC_vs_ctrl", 
                 add_names = TRUE,
                additional_title = "pY")
```

```{r, fig.width=10, fig.height=10}
Return_DEP_Hits_Plots(data = pY_Set1_form, dep_EBC_vs_ctrl_pY, comparison = "EBC_vs_ctrl_diff")
```


#### EC vs E
```{r}
Test_PhosphoData(pY_Set1_form, comparison = "EC", comparison_base = "E") %>% GGPlotly_Volcano_Test

data_diff_EC_vs_E_pY <- test_diff(pY_se_Set1, type = "manual", 
                              test = c("EC_vs_E"))
dep_EC_vs_E_pY <- add_rejections_SH(data_diff_EC_vs_E_pY, alpha = 0.05, lfc = log2(1.2))
GGPlotly_Volcano(dep_EC_vs_E_pY, contrast = "EC_vs_E",  add_names = TRUE, additional_title = "pY", proteins_of_interest = "EGFR")
```

```{r, fig.width=10, fig.height=10}
Return_DEP_Hits_Plots(data = pY_Set1_form, dep_EC_vs_E_pY, comparison = "EC_vs_E_diff")

#data_results <- get_df_long(dep)
```

#### EBC vs EC
```{r}
Test_PhosphoData(pY_Set1_form, comparison = "EBC", comparison_base = "EC") %>% GGPlotly_Volcano_Test

data_diff_EBC_vs_EC_pY <- test_diff(pY_se_Set1, type = "manual", 
                              test = c("EBC_vs_EC"))
dep_EBC_vs_EC_pY <- add_rejections_SH(data_diff_EBC_vs_EC_pY, alpha = 0.05, lfc = log2(1.2))
GGPlotly_Volcano(dep_EBC_vs_EC_pY, contrast = "EBC_vs_EC",  add_names = TRUE, additional_title = "pY")
```

```{r, fig.width=10, fig.height=10}
Return_DEP_Hits_Plots(data = pY_Set1_form, dep_EBC_vs_EC_pY, comparison = "EBC_vs_EC_diff")

#data_results <- get_df_long(dep)
```


### Serine/Threonine fractionated
#### Each condition vs ctrl
```{r, eval=TRUE}
#Test_PhosphoData(pST_Set1_form, comparison = "E", comparison_base = "ctrl") %>% GGPlotly_Volcano_Test

data_diff_E_vs_ctrl_pST <- test_diff(pST_se_Set1, type="manual", test = "E_vs_ctrl")
dep_E_vs_ctrl_pST <- add_rejections_SH(data_diff_E_vs_ctrl_pST, alpha = 0.05, lfc = log2(1.2))
GGPlotly_Volcano(dep_E_vs_ctrl_pST, contrast = "E_vs_ctrl", 
                 add_names = TRUE,
                additional_title = "pST") 
```

```{r, fig.width=10, fig.height=10}
Return_DEP_Hits_Plots(data = pST_Set1_form, dep_E_vs_ctrl_pST, comparison = "E_vs_ctrl_diff")
```

```{r}
#Test_PhosphoData(pST_Set1_form, comparison = "EC", comparison_base = "ctrl") %>% GGPlotly_Volcano_Test

data_diff_EC_vs_ctrl_pST <- test_diff(pST_se_Set1, type="manual", test = "EC_vs_ctrl")
dep_EC_vs_ctrl_pST <- add_rejections_SH(data_diff_EC_vs_ctrl_pST, alpha = 0.05, lfc = log2(1.2))
GGPlotly_Volcano(dep_EC_vs_ctrl_pST, contrast = "EC_vs_ctrl", 
                 add_names = TRUE,
                additional_title = "pST") 
```

```{r, fig.width=10, fig.height=10}
Return_DEP_Hits_Plots(data = pST_Set1_form, dep_EC_vs_ctrl_pST, comparison = "EC_vs_ctrl_diff")
```

```{r, eval=FALSE}
Plot_Enrichment_Single_Pathway(dep_EC_vs_ctrl_pST, comparison = "EC_vs_ctrl_diff", 
                               pw = "Epigenetic regulation of gene expression")
```

```{r}
#Test_PhosphoData(pST_Set1_form, comparison = "EBC", comparison_base = "ctrl") %>% GGPlotly_Volcano_Test

data_diff_EBC_vs_ctrl_pST <- test_diff(pST_se_Set1, type="manual", test = "EBC_vs_ctrl")
dep_EBC_vs_ctrl_pST <- add_rejections_SH(data_diff_EBC_vs_ctrl_pST, alpha = 0.05, lfc = log2(1.2))
GGPlotly_Volcano(dep_EBC_vs_ctrl_pST, contrast = "EBC_vs_ctrl", 
                 add_names = TRUE,
                additional_title = "pST")
```

```{r, fig.width=10, fig.height=10}
Return_DEP_Hits_Plots(data = pST_Set1_form, dep_EBC_vs_ctrl_pST, comparison = "EBC_vs_ctrl_diff")
```


#### EC vs E
```{r}
#Test_PhosphoData(pST_Set1_form, comparison = "EC", comparison_base = "E") %>% GGPlotly_Volcano_Test

data_diff_EC_vs_E_pST <- test_diff(pST_se_Set1, type = "manual", 
                              test = c("EC_vs_E"))
dep_EC_vs_E_pST <- add_rejections_SH(data_diff_EC_vs_E_pST, alpha = 0.05, lfc = log2(1.2))
GGPlotly_Volcano(dep_EC_vs_E_pST, contrast = "EC_vs_E",  add_names = TRUE, additional_title = "pST", proteins_of_interest = "EGFR")
```

```{r, fig.width=10, fig.height=10}
Return_DEP_Hits_Plots(data = pST_Set1_form, dep_EC_vs_E_pST, comparison = "EC_vs_E_diff")

#data_results <- get_df_long(dep)
```

#### EBC vs EC
```{r}
#Test_PhosphoData(pST_Set1_form, comparison = "EBC", comparison_base = "EC") %>% GGPlotly_Volcano_Test

data_diff_EBC_vs_EC_pST <- test_diff(pST_se_Set1, type = "manual", 
                              test = c("EBC_vs_EC"))
dep_EBC_vs_EC_pST <- add_rejections_SH(data_diff_EBC_vs_EC_pST, alpha = 0.05, lfc = log2(1.2))
GGPlotly_Volcano(dep_EBC_vs_EC_pST, contrast = "EBC_vs_EC",  add_names = TRUE, additional_title = "pST")
```

```{r, fig.width=10, fig.height=10}
Return_DEP_Hits_Plots(data = pST_Set1_form, dep_EBC_vs_EC_pST, comparison = "EBC_vs_EC_diff")

#data_results <- get_df_long(dep)
```


### Serine/Threonine one shot
#### Each condition vs ctrl
```{r, eval=TRUE}
#Test_PhosphoData(pST_Set1_oneshot_form, comparison = "E", comparison_base = "ctrl") %>% GGPlotly_Volcano_Test

data_diff_E_vs_ctrl_pST_oneshot <- test_diff(pST_se_Set1_oneshot, type="manual", test = "E_vs_ctrl")
dep_E_vs_ctrl_pST_oneshot <- add_rejections_SH(data_diff_E_vs_ctrl_pST_oneshot, alpha = 0.05, lfc = log2(1.2))
GGPlotly_Volcano(dep_E_vs_ctrl_pST_oneshot, contrast = "E_vs_ctrl", 
                 add_names = TRUE,
                additional_title = "pST") 
```

```{r, fig.width=10, fig.height=10}
Return_DEP_Hits_Plots(data = pST_Set1_oneshot_form, dep_E_vs_ctrl_pST_oneshot, comparison = "E_vs_ctrl_diff")
```

```{r}
#Test_PhosphoData(pST_Set1_oneshot_form, comparison = "EC", comparison_base = "ctrl") %>% GGPlotly_Volcano_Test

data_diff_EC_vs_ctrl_pST_oneshot <- test_diff(pST_se_Set1_oneshot, type="manual", test = "EC_vs_ctrl")
dep_EC_vs_ctrl_pST_oneshot <- add_rejections_SH(data_diff_EC_vs_ctrl_pST_oneshot, alpha = 0.05, lfc = log2(1.2))
GGPlotly_Volcano(dep_EC_vs_ctrl_pST_oneshot, contrast = "EC_vs_ctrl", 
                 add_names = TRUE,
                additional_title = "pST") 
```

```{r, fig.width=10, fig.height=10}
Return_DEP_Hits_Plots(data = pST_Set1_oneshot_form, dep_EC_vs_ctrl_pST_oneshot, comparison = "EC_vs_ctrl_diff")
```

```{r, eval=FALSE}
Plot_Enrichment_Single_Pathway(dep_EC_vs_ctrl_pST_oneshot, comparison = "EC_vs_ctrl_diff", 
                               pw = "Epigenetic regulation of gene expression")
```

```{r}
#Test_PhosphoData(pST_Set1_oneshot_form, comparison = "EBC", comparison_base = "ctrl") %>% GGPlotly_Volcano_Test

data_diff_EBC_vs_ctrl_pST_oneshot <- test_diff(pST_se_Set1_oneshot, type="manual", test = "EBC_vs_ctrl")
dep_EBC_vs_ctrl_pST_oneshot <- add_rejections_SH(data_diff_EBC_vs_ctrl_pST_oneshot, alpha = 0.05, lfc = log2(1.2))
GGPlotly_Volcano(dep_EBC_vs_ctrl_pST_oneshot, contrast = "EBC_vs_ctrl", 
                 add_names = TRUE,
                additional_title = "pST")
```

```{r, fig.width=10, fig.height=10}
Return_DEP_Hits_Plots(data = pST_Set1_oneshot_form, dep_EBC_vs_ctrl_pST_oneshot, comparison = "EBC_vs_ctrl_diff")
```


#### EC vs E
```{r}
#Test_PhosphoData(pST_Set1_oneshot_form, comparison = "EC", comparison_base = "E") %>% GGPlotly_Volcano_Test

data_diff_EC_vs_E_pST_oneshot <- test_diff(pST_se_Set1_oneshot, type = "manual", 
                              test = c("EC_vs_E"))
dep_EC_vs_E_pST_oneshot <- add_rejections_SH(data_diff_EC_vs_E_pST_oneshot, alpha = 0.05, lfc = log2(1.2))
GGPlotly_Volcano(dep_EC_vs_E_pST_oneshot, contrast = "EC_vs_E",  add_names = TRUE, additional_title = "pST", proteins_of_interest = "EGFR")
```

```{r, fig.width=10, fig.height=10}
Return_DEP_Hits_Plots(data = pST_Set1_oneshot_form, dep_EC_vs_E_pST_oneshot, comparison = "EC_vs_E_diff")

#data_results <- get_df_long(dep)
```

#### EBC vs EC
```{r}
#Test_PhosphoData(pST_Set1_oneshot_form, comparison = "EBC", comparison_base = "EC") %>% GGPlotly_Volcano_Test

data_diff_EBC_vs_EC_pST_oneshot <- test_diff(pST_se_Set1_oneshot, type = "manual", 
                              test = c("EBC_vs_EC"))
dep_EBC_vs_EC_pST_oneshot <- add_rejections_SH(data_diff_EBC_vs_EC_pST_oneshot, alpha = 0.05, lfc = log2(1.2))
GGPlotly_Volcano(dep_EBC_vs_EC_pST_oneshot, contrast = "EBC_vs_EC",  add_names = TRUE, additional_title = "pST")
```

```{r, fig.width=10, fig.height=10}
Return_DEP_Hits_Plots(data = pST_Set1_oneshot_form, dep_EBC_vs_EC_pST_oneshot, comparison = "EBC_vs_EC_diff")

#data_results <- get_df_long(dep)
```



## Motif enrichment analysis
### 7AA Sequences pST fractionated
```{r}
all_pST_sites <- (rowData(dep_E_vs_ctrl_pST) %>% as_tibble()) %>% select( Annotated_Sequence, HGNC_Symbol, Master.Protein.Accessions) %>% unique

table(all_pST_sites$Master.Protein.Accessions %in% Phosphorylation_site_dataset$ACC_ID )

Phosphorylation_site_dataset_pST <- Phosphorylation_site_dataset %>% filter(ACC_ID %in% all_pST_sites$Master.Protein.Accessions)

# TODO: s/t in first place might have TMT and or phos
# TODO did not account for double phosphorylated peptides
extracted_trimmed_peptide <- function(peptide){
  str_extract(peptide, "s|t")
  
  if( nchar( sub("[ts].*", "", peptide )) >= 7 & 
      nchar( sub(".*[ts]", "", peptide )) >= 7){
    pep_trimmed <- str_extract(peptide, ".......[ts].......")
  }else if(nchar( sub("[ts].*", "", peptide )) < 7 &
           nchar( sub(".*[ts]", "", peptide )) < 7){
    pep_trimmed <- str_extract(peptide, ".*[ts].*")
  }else if(nchar( sub("[ts].*", "", peptide )) < 7){
    pep_trimmed <- str_extract(peptide, ".*[ts].......")
  }else if(nchar( sub(".*[ts]", "", peptide )) < 7){
    pep_trimmed <- str_extract(peptide, ".......[ts].*")
  }else{stop(paste("Could not extract trimmed sequenxce from", peptide )) }
  return(pep_trimmed)
}

all_pST_sites$trimmed_sequence <- sapply(all_pST_sites$Annotated_Sequence, extracted_trimmed_peptide, simplify = T)
 
get_plusminus_7AA_sequence <- function(peptide, protein){
  mod_peptide <- paste0(str_to_upper(sub("[ts].*", "", peptide) ),
         str_extract(peptide, "[ts]"),
         str_to_upper(sub(".*[ts]", "", peptide) ) )
  sequ <- Phosphorylation_site_dataset_pST %>% filter(ACC_ID == protein) %>%
    mutate( SEQUENCE = paste0(str_to_upper(str_sub(`SITE_+/-7_AA`, 1, 7)),
                               str_sub(`SITE_+/-7_AA`, 8,8),
                              paste0(str_to_upper(str_sub(`SITE_+/-7_AA`, 9, 15)))) ) %>% 
    filter(str_detect(SEQUENCE, mod_peptide)) %>% .$SEQUENCE
  if(length(sequ) == 0){
    return(NA)}else{return(sequ)}
}

sequences7AA <- sapply(1:nrow(all_pST_sites), function(i){
  sequ <- get_plusminus_7AA_sequence(peptide = all_pST_sites$trimmed_sequence[i],
                             protein = all_pST_sites$Master.Protein.Accessions[i]) %>% unique
  if(length(sequ) != 1){
    return(NA)}else{return(sequ)}
  }, simplify = T) 

table(is.na(sequences7AA) )

all_pST_sites$Sequence_7_AA <- sequences7AA
```

### 7AA Sequences pST one shot
```{r}
all_pST_oneshot_sites <- (rowData(dep_E_vs_ctrl_pST_oneshot) %>% as_tibble()) %>% select( Annotated_Sequence, HGNC_Symbol, Master.Protein.Accessions) %>% unique

table(all_pST_oneshot_sites$Master.Protein.Accessions %in% Phosphorylation_site_dataset$ACC_ID )

Phosphorylation_site_dataset_pST_oneshot <- Phosphorylation_site_dataset %>% filter(ACC_ID %in% all_pST_oneshot_sites$Master.Protein.Accessions)

# TODO: s/t in first place might have TMT and or phos
# TODO did not account for double phosphorylated peptides
extracted_trimmed_peptide <- function(peptide){
  str_extract(peptide, "s|t")
  
  if( nchar( sub("[ts].*", "", peptide )) >= 7 & 
      nchar( sub(".*[ts]", "", peptide )) >= 7){
    pep_trimmed <- str_extract(peptide, ".......[ts].......")
  }else if(nchar( sub("[ts].*", "", peptide )) < 7 &
           nchar( sub(".*[ts]", "", peptide )) < 7){
    pep_trimmed <- str_extract(peptide, ".*[ts].*")
  }else if(nchar( sub("[ts].*", "", peptide )) < 7){
    pep_trimmed <- str_extract(peptide, ".*[ts].......")
  }else if(nchar( sub(".*[ts]", "", peptide )) < 7){
    pep_trimmed <- str_extract(peptide, ".......[ts].*")
  }else{stop(paste("Could not extract trimmed sequenxce from", peptide )) }
  return(pep_trimmed)
}

all_pST_oneshot_sites$trimmed_sequence <- sapply(all_pST_oneshot_sites$Annotated_Sequence, extracted_trimmed_peptide, simplify = T)
 
get_plusminus_7AA_sequence <- function(peptide, protein){
  mod_peptide <- paste0(str_to_upper(sub("[ts].*", "", peptide) ),
         str_extract(peptide, "[ts]"),
         str_to_upper(sub(".*[ts]", "", peptide) ) )
  sequ <- Phosphorylation_site_dataset_pST_oneshot %>% filter(ACC_ID == protein) %>%
    mutate( SEQUENCE = paste0(str_to_upper(str_sub(`SITE_+/-7_AA`, 1, 7)),
                               str_sub(`SITE_+/-7_AA`, 8,8),
                              paste0(str_to_upper(str_sub(`SITE_+/-7_AA`, 9, 15)))) ) %>% 
    filter(str_detect(SEQUENCE, mod_peptide)) %>% .$SEQUENCE
  if(length(sequ) == 0){
    return(NA)}else{return(sequ)}
}

sequences7AA <- sapply(1:nrow(all_pST_oneshot_sites), function(i){
  sequ <- get_plusminus_7AA_sequence(peptide = all_pST_oneshot_sites$trimmed_sequence[i],
                             protein = all_pST_oneshot_sites$Master.Protein.Accessions[i]) %>% unique
  if(length(sequ) != 1){
    return(NA)}else{return(sequ)}
  }, simplify = T) 

table(is.na(sequences7AA) )

all_pST_oneshot_sites$Sequence_7_AA <- sequences7AA
```

### E vs ctrl
```{r}
E_vs_ctrl_pST_7AA <- left_join( (rowData(dep_E_vs_ctrl_pST) %>% as_tibble() %>% 
    select(annotation = ID, Annotated_Sequence, HGNC_Symbol, 
           fc = E_vs_ctrl_diff,  
           p = E_vs_ctrl_p.adj) 
    %>% unique ),
  (all_pST_sites %>% select(Annotated_Sequence, Sequence_7_AA, HGNC_Symbol)),
  by=c("Annotated_Sequence", "HGNC_Symbol") ) %>% filter(!is.na(Sequence_7_AA)) %>% 
  mutate( peptide = str_to_upper(Sequence_7_AA) ) %>%
  select(annotation, peptide, fc, p) %>% as.data.frame()


E_vs_ctrl_pST_7AA %>% select(peptide, fc, p)  %>% write.table(file = "For_motif_analysis/E_vs_ctrl_pST.txt", quote = FALSE, row.names = F, col.names = F, sep = "\t")
```

### E vs ctrl on shot
```{r}
E_vs_ctrl_pST_oneshot_7AA <- left_join( (rowData(dep_E_vs_ctrl_pST_oneshot) %>% as_tibble() %>% 
    select(annotation = ID, Annotated_Sequence, HGNC_Symbol, 
           fc = E_vs_ctrl_diff,  
           p = E_vs_ctrl_p.adj) 
    %>% unique ),
  (all_pST_oneshot_sites %>% select(Annotated_Sequence, Sequence_7_AA, HGNC_Symbol)),
  by=c("Annotated_Sequence", "HGNC_Symbol") ) %>% filter(!is.na(Sequence_7_AA)) %>% 
  mutate( peptide = str_to_upper(Sequence_7_AA) ) %>%
  select(annotation, peptide, fc, p) %>% as.data.frame()


E_vs_ctrl_pST_oneshot_7AA %>% select(peptide, fc, p)  %>% write.table(file = "For_motif_analysis/E_vs_ctrl_pST_oneshot.txt", quote = FALSE, row.names = F, col.names = F, sep = "\t")
```

### EC vs ctrl
```{r}
EC_vs_ctrl_pST_7AA <- left_join( (rowData(dep_EC_vs_ctrl_pST) %>% as_tibble() %>% 
    select(annotation = ID, Annotated_Sequence, HGNC_Symbol, 
           fc = EC_vs_ctrl_diff,  
           p = EC_vs_ctrl_p.adj) 
    %>% unique ),
  (all_pST_sites %>% select(Annotated_Sequence, Sequence_7_AA, HGNC_Symbol)),
  by=c("Annotated_Sequence", "HGNC_Symbol") ) %>% filter(!is.na(Sequence_7_AA)) %>% 
  mutate( peptide = str_to_upper(Sequence_7_AA) ) %>%
  select(annotation, peptide, fc, p) %>% as.data.frame()


EC_vs_ctrl_pST_7AA %>% select(peptide, fc, p)  %>% write.table(file = "For_motif_analysis/EC_vs_ctrl_pST.txt", quote = FALSE, row.names = F, col.names = F, sep = "\t")
```

### EC vs ctrl on shot
```{r}
EC_vs_ctrl_pST_oneshot_7AA <- left_join( (rowData(dep_EC_vs_ctrl_pST_oneshot) %>% as_tibble() %>% 
    select(annotation = ID, Annotated_Sequence, HGNC_Symbol, 
           fc = EC_vs_ctrl_diff,  
           p = EC_vs_ctrl_p.adj) 
    %>% unique ),
  (all_pST_oneshot_sites %>% select(Annotated_Sequence, Sequence_7_AA, HGNC_Symbol)),
  by=c("Annotated_Sequence", "HGNC_Symbol") ) %>% filter(!is.na(Sequence_7_AA)) %>% 
  mutate( peptide = str_to_upper(Sequence_7_AA) ) %>%
  select(annotation, peptide, fc, p) %>% as.data.frame()


EC_vs_ctrl_pST_oneshot_7AA %>% select(peptide, fc, p)  %>% write.table(file = "For_motif_analysis/EC_vs_ctrl_pST_oneshot.txt", quote = FALSE, row.names = F, col.names = F, sep = "\t")
```

### EC vs E
```{r}
EC_vs_E_pST_7AA <- left_join( (rowData(dep_EC_vs_E_pST) %>% as_tibble() %>% 
    select(annotation = ID, Annotated_Sequence, HGNC_Symbol, 
           fc = EC_vs_E_diff,  
           p = EC_vs_E_p.adj) 
    %>% unique ),
  (all_pST_sites %>% select(Annotated_Sequence, Sequence_7_AA, HGNC_Symbol)),
  by=c("Annotated_Sequence", "HGNC_Symbol") ) %>% filter(!is.na(Sequence_7_AA)) %>% 
  mutate( peptide = str_to_upper(Sequence_7_AA) ) %>%
  select(annotation, peptide, fc, p) %>% as.data.frame()


EC_vs_E_pST_7AA %>% select(peptide, fc, p)  %>% write.table(file = "For_motif_analysis/EC_vs_E_pST.txt", quote = FALSE, row.names = F, col.names = F, sep = "\t")
```

### EC vs E on shot
```{r}
EC_vs_E_pST_oneshot_7AA <- left_join( (rowData(dep_EC_vs_E_pST_oneshot) %>% as_tibble() %>% 
    select(annotation = ID, Annotated_Sequence, HGNC_Symbol, 
           fc = EC_vs_E_diff,  
           p = EC_vs_E_p.adj) 
    %>% unique ),
  (all_pST_oneshot_sites %>% select(Annotated_Sequence, Sequence_7_AA, HGNC_Symbol)),
  by=c("Annotated_Sequence", "HGNC_Symbol") ) %>% filter(!is.na(Sequence_7_AA)) %>% 
  mutate( peptide = str_to_upper(Sequence_7_AA) ) %>%
  select(annotation, peptide, fc, p) %>% as.data.frame()


EC_vs_E_pST_oneshot_7AA %>% select(peptide, fc, p)  %>% write.table(file = "For_motif_analysis/EC_vs_E_pST_oneshot.txt", quote = FALSE, row.names = F, col.names = F, sep = "\t")
```

### EBC vs ctrl
```{r}
EBC_vs_ctrl_pST_7AA <- left_join( (rowData(dep_EBC_vs_ctrl_pST) %>% as_tibble() %>% 
    select(annotation = ID, Annotated_Sequence, HGNC_Symbol, 
           fc = EBC_vs_ctrl_diff,  
           p = EBC_vs_ctrl_p.adj) 
    %>% unique ),
  (all_pST_sites %>% select(Annotated_Sequence, Sequence_7_AA, HGNC_Symbol)),
  by=c("Annotated_Sequence", "HGNC_Symbol") ) %>% filter(!is.na(Sequence_7_AA)) %>% 
  mutate( peptide = str_to_upper(Sequence_7_AA) ) %>%
  select(annotation, peptide, fc, p) %>% as.data.frame()


EBC_vs_ctrl_pST_7AA %>% select(peptide, fc, p)  %>% write.table(file = "For_motif_analysis/EBC_vs_ctrl_pST.txt", quote = FALSE, row.names = F, col.names = F, sep = "\t")
```

### EBC vs ctrl on shot
```{r}
EBC_vs_ctrl_pST_oneshot_7AA <- left_join( (rowData(dep_EBC_vs_ctrl_pST_oneshot) %>% as_tibble() %>% 
    select(annotation = ID, Annotated_Sequence, HGNC_Symbol, 
           fc = EBC_vs_ctrl_diff,  
           p = EBC_vs_ctrl_p.adj) 
    %>% unique ),
  (all_pST_oneshot_sites %>% select(Annotated_Sequence, Sequence_7_AA, HGNC_Symbol)),
  by=c("Annotated_Sequence", "HGNC_Symbol") ) %>% filter(!is.na(Sequence_7_AA)) %>% 
  mutate( peptide = str_to_upper(Sequence_7_AA) ) %>%
  select(annotation, peptide, fc, p) %>% as.data.frame()


EBC_vs_ctrl_pST_oneshot_7AA %>% select(peptide, fc, p)  %>% write.table(file = "For_motif_analysis/EBC_vs_ctrl_pST_oneshot.txt", quote = FALSE, row.names = F, col.names = F, sep = "\t")
```


### EBC vs EC
```{r}
EBC_vs_EC_pST_7AA <- left_join( (rowData(dep_EBC_vs_EC_pST) %>% as_tibble() %>% 
    select(annotation = ID, Annotated_Sequence, HGNC_Symbol, 
           fc = EBC_vs_EC_diff,  
           p = EBC_vs_EC_p.adj) 
    %>% unique ),
  (all_pST_sites %>% select(Annotated_Sequence, Sequence_7_AA, HGNC_Symbol)),
  by=c("Annotated_Sequence", "HGNC_Symbol") ) %>% filter(!is.na(Sequence_7_AA)) %>% 
  mutate( peptide = str_to_upper(Sequence_7_AA) ) %>%
  select(annotation, peptide, fc, p) %>% as.data.frame()


EBC_vs_EC_pST_7AA %>% select(peptide, fc, p)  %>% write.table(file = "For_motif_analysis/EBC_vs_EC_pST.txt", quote = FALSE, row.names = F, col.names = F, sep = "\t")
```

### EBC vs EC on shot
```{r}
EBC_vs_EC_pST_oneshot_7AA <- left_join( (rowData(dep_EBC_vs_EC_pST_oneshot) %>% as_tibble() %>% 
    select(annotation = ID, Annotated_Sequence, HGNC_Symbol, 
           fc = EBC_vs_EC_diff,  
           p = EBC_vs_EC_p.adj) 
    %>% unique ),
  (all_pST_oneshot_sites %>% select(Annotated_Sequence, Sequence_7_AA, HGNC_Symbol)),
  by=c("Annotated_Sequence", "HGNC_Symbol") ) %>% filter(!is.na(Sequence_7_AA)) %>% 
  mutate( peptide = str_to_upper(Sequence_7_AA) ) %>%
  select(annotation, peptide, fc, p) %>% as.data.frame()


EBC_vs_EC_pST_oneshot_7AA %>% select(peptide, fc, p)  %>% write.table(file = "For_motif_analysis/EBC_vs_EC_pST_oneshot.txt", quote = FALSE, row.names = F, col.names = F, sep = "\t")
```

# Session Info
```{r}
sessionInfo()
knitr::knit_exit()
```
